.circleci/config.yml

```diff
@@ -65,6 +65,23 @@ jobs:
       - run:
           name: Release
           command: npm run release
+  
+  storybook:
+    executor: main-executor
+    steps:
+      - checkout_from_workspace
+      - run:
+          name: Build Storybook
+          command: npm run storybook:ci
+      - run:
+          name: Publish Storybook
+          command: |
+            # Check if there's anything to commit before committing.
```

Comment:
Is there a reason why you didn't stick with Storybook deployer? You [can specify the output directory](https://github.com/storybookjs/storybook-deployer#configure-build-directory), which i thought is what we wanted to change. I would prefer to use the tools provided by the tool creator

Reply:
The tool creates a new gh-pages branch and does a lot of magic along with it to get gh-pages to work with storybook. The gh-pages branch setup usually erases everything from the repo and leaves just the [storybook] stuff thats only going to be served for github pages.

The new approach just creates a static version of the app (link in my review) and creates it as a subdirectory in the `docs/` folder. Allowing us to make `docs/` itself have it's own static setup and other subdirectories for other documentation related setups.
This setup is the same as the SDK, which uses the `docs/` folder to as the source for github pages.

The `master branch docs/ folder` serving github pages is a relatively new setting (a little over a year old) to setup github pages. The deployer's approach is the old way, and complicates things using branches.

Both are tools provided by the creator, just this new approach makes committing a manual step. And this step was a way to make sure git didn't exit 1 on circleci if there was nothing to commit. The SDK was also setup to commit doc changes first, but Tim's setup is that it's own job last (which also makes sure that CI gets skipped (like semantic-release), so if docs fails at some point you can `retry from fail job` and nothing changes for publishing.

Reply:
OK, the only thing I'm worried about is the check, `git status --procelain`,I feel like it's too general. There could be a previous step that generates stuff outside Storybook and then commits those files unintended

Reply:
There should be nothing to commit but changes made/generated by Storybook, since the only command ran in the storybook job is `storybook:ci`. there shouldn't be anything before that since it's a new, "clean" job in the circleci's workflow

Reply:
> OK, the only thing I'm worried about is the check, `git status --procelain`,I feel like it's too general.

@InteractiveTimmy, why don't you just `diff` it on a specific directory then?

Reply:
Find a more elegant solution to this. Will update when ready.

Reply:
This has been implemented, please review 👍

Reply:
Thanks for looking into it @InteractiveTimmy  🙂

Reply:
I still suggest we use `porcelain`
It’s a more script friendly, “LTS” version of `short`

https://git-scm.com/docs/git-status#Documentation/git-status.txt---porcelainltversiongt

Reply:
Looks like we're set on porcelain. Pushed and ready for re-review.